addons/common/functions/fnc_uniqueItems.sqf

```diff
@@ -1,37 +1,84 @@
 #include "..\script_component.hpp"
 /*
- * Author: mharis001
- * Returns list of unique items in a unit's inventory.
- * Items are cached if unit is ACE_player.
+ * Author: mharis001, Blue, Brett Mayson
+ * Returns list of unique items in the target's inventory.
  *
  * Arguments:
- * 0: Unit <OBJECT>
+ * 0: Target <OBJECT>
+ * 1: Include magazines <NUMBER>
+ *   0: No (default)
+ *   1: Yes
+ *   2: Only magazines
  *
  * Return Value:
  * Items <ARRAY>
  *
  * Example:
- * [player] call ace_common_fnc_uniqueItems
+ * [player, 2] call ace_common_fnc_getUniqueItems
  *
  * Public: No
  */
 
-params ["_unit"];
+params ["_target", ["_includeMagazines", 0]];
 
 private _fnc_getItems = {
-    private _items = (getItemCargo uniformContainer _unit) select 0;
-    _items append ((getItemCargo vestContainer _unit) select 0);
-    _items append ((getItemCargo backpackContainer _unit) select 0);
+    private _items = [];
+
+    private _inventoryItems = [];
+
+    _inventoryItems append ((getItemCargo uniformContainer _target) select 0);
+    _inventoryItems append ((getItemCargo vestContainer _target) select 0);
+    _inventoryItems append ((getItemCargo backpackContainer _target) select 0);
+
+    _items set [0, _inventoryItems];
+    _items set [1, (magazines _target)];
 
     _items arrayIntersect _items
 };
 
-// Use cached items list if unit is ACE_player
-if (_unit isEqualTo ACE_player) then {
+// Cache items list if unit is ACE_player
+if (_target isEqualTo ACE_player) then {
     if (isNil QGVAR(uniqueItemsCache)) then {
         GVAR(uniqueItemsCache) = call _fnc_getItems;
     };
-    +GVAR(uniqueItemsCache)
+
+    switch (_includeMagazines) do {
+        case 0: {
+            GVAR(uniqueItemsCache) select 0;
+        };
+        case 1: {
+            (GVAR(uniqueItemsCache) select 1) + (GVAR(uniqueItemsCache) select 0);
+        };
+        case 2: {
+            GVAR(uniqueItemsCache) select 1;
+        };
+    };
 } else {
-    call _fnc_getItems;
+    if (_isVehicle) then {
```

Comment:
This is undefined.

What is the point of looking at vehicles? We don't care about vehicles here, as far as I can tell.